SOOP/SOOP_XBT/DELAYED/xbt_dm_imos_conversion.py

```diff
@@ -469,33 +485,33 @@ def adjust_position_qc_flags(annex, data):
     #AW change distinguish between PE+LALO - flag =4 (position fail) and PE+LATI|LONG - flag 2 (position corrected)
     #AW we also should also set the time QC flag to 4 for date-time failures see func adjust_time_qc_flags() below
     #print("Annex=",annex)
-    if 'PE' in annex['act_code']:
+    if 'PE' in annex['act_code'] and not '5' in data['LONGITUDE_quality_control']:
         if ('LATI' in annex['act_parm']) or ('LONG' in annex['act_parm']):
             #print("annex['act_code']1=",annex['act_code'])
             #print("annex['act_parm']1=",annex['act_parm'])
-            LOGGER.info('Position correction (PEA) in original file, changing position flags to level 2.')
+            LOGGER.info('Position correction (PEA) in original file, changing position flags to level 5.')
 
-            data['LATITUDE_quality_control'] = 2
-            data['LONGITUDE_quality_control'] = 2
-        if 'LALO' in annex['act_parm']:
+            data['LATITUDE_quality_control'] = 5
+            data['LONGITUDE_quality_control'] = 5
+        if 'LALO' in annex['act_parm'] and not '3' in data['LONGITUDE_quality_control']:
 
-            LOGGER.info('Position failure (PER) in original file, changing position flags to level 4.')
-            data['LATITUDE_quality_control'] = 4
-            data['LONGITUDE_quality_control'] = 4
+            LOGGER.info('Position failure (PER) in original file, changing position flags to level 3.')
+            data['LATITUDE_quality_control'] = 3
+            data['LONGITUDE_quality_control'] = 3
     
     return data
 
 def adjust_time_qc_flags(annex, data):
     #AW Add function  we also should also set the time QC flag to 4 for date-time failures TE in annex['act_code'] + DATI in annex['act_parm']
-    #or set time QC to flag 2 if date/time has been corrected
+    #or set time QC to flag 5 if date/time has been corrected
     #print("Annex=",annex)
-    if 'TE' in annex['act_code'] and 'DATI' in annex['act_parm']:
-        LOGGER.info('Date-Time failure (TER) in original file, setting time qc flag to level 4.')
-        data['TIME_quality_control'] = 4
+    if 'TE' in annex['act_code'] and 'DATI' in annex['act_parm'] and not '3' in data['TIME_quality_control']:
+        LOGGER.info('Date-Time failure (TER) in original file, setting time qc flag to level 3.')
+        data['TIME_quality_control'] = 3
         
-    if 'TE' in annex['act_code'] and ('TIME' in annex['act_parm'] or 'DATE' in annex['act_parm']):
-        LOGGER.info('Date and/or Time has been corrected (TEA) in original file, setting time qc flag to level 2.')
-        data['TIME_quality_control'] = 2
+    if 'TE' in annex['act_code'] and ('TIME' in annex['act_parm'] or 'DATE' in annex['act_parm']) and not '5' in data['TIME_quality_control']:
```

Comment:
```python
    if 'TE' in annex['act_code'] and ('TIME' in annex['act_parm'] or 'DATE' in annex['act_parm']) and not data['TIME_quality_control'] == 5:
```

Reply:
done, will need to test on a file with TE flag.

Reply:
Missed one of these, recommitted.